src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -2950,7 +2959,16 @@ private void findSocketUsingThreading(InetAddress[] inetAddrs, int portNumber,
                     if (timeRemaining <= 0 || (!result.equals(Result.UNKNOWN)))
                         break;
 
-                    parentCondition.await(timeRemaining, TimeUnit.MILLISECONDS);
+                    try {
+                        parentCondition.await(timeRemaining, TimeUnit.MILLISECONDS);
+                    } catch (InterruptedException ie) {
+                        // This uncaught exception was what was casuing the issues we saw previously with many
+                        // threads being created
+                        if (logger.isLoggable(Level.FINER)) {
+                            logger.finer(this.toString() + " Current thread has been interrupted.");
+                        }
+                        Thread.currentThread().interrupt();
```

Comment:
The log isn't necessary and the interrupt() will leave the driver in the same state we are trying to prevent with the catch...
```suggestion
```

Reply:
I thought the issue was with that the uncaught exception was causing problems, not that the thread is being interrupted.

Reply:
The issue is that the thread has been interrupted (that's the signal that is used to get parentCondition.await() to continue), unless we specifically catch the interrupt here (catch InterruptedException), which resets the thread interrupted status and keeps it from propagating further. If we don't catch (and thus reset) the InteruptedException, all subsequent Thread.sleep() calls just fall through. That skipping of all the sleeps is what triggers the thread storm since the outer connection retries no longer pause in between each try.

Reply:
I see the original comment I proposed is completely different now, too. It should be changed back. We aren't ignoring the interrupt. We are expecting it and clearing it via the catch block.